From 91d77dda698cba0668b03f867fa7f7205dd8459e Mon Sep 17 00:00:00 2001
From: dragonGR <alex@dragongr.dev>
Date: Wed, 20 Jul 2022 17:58:20 +0300
Subject: [PATCH] PlaylistFetcher: Speed up the network buffer

This bool will result to a better network buffering.

Nowadays the world is equipped with decent modems/routers.
The days where we had 802.11b only are long gone and we're not coming back to it.
Why have slow buffer when your network is fast enough? Speed that up!

Change-Id: Ic92e5192eb282dde288290c9003dc10e724a3f77
Signed-off-by: Alex Tsanis <alex@dragongr.dev>
---
 media/libstagefright/httplive/PlaylistFetcher.cpp | 13 +++++++++++++
 media/libstagefright/httplive/PlaylistFetcher.h   |  2 ++
 2 files changed, 15 insertions(+)

diff --git a/media/libstagefright/httplive/PlaylistFetcher.cpp b/media/libstagefright/httplive/PlaylistFetcher.cpp
index c311f146bd..b7e2647fd3 100644
--- a/media/libstagefright/httplive/PlaylistFetcher.cpp
+++ b/media/libstagefright/httplive/PlaylistFetcher.cpp
@@ -2259,4 +2259,17 @@ void PlaylistFetcher::updateTargetDuration() {
     msg->post();
 }
 
+bool PlaylistFetcher::bufferAvailable() const {
+    status_t result = OK;
+    for (size_t i = 0; i < mPacketSources.size(); ++i) {
+        if ((mStreamTypeMask & mPacketSources.keyAt(i)) == 0) {
+            continue;
+        }
+        if (mPacketSources.valueAt(i)->hasBufferAvailable(&result)) {
+            return true;
+        }
+    }
+    return false;
+}
+
 }  // namespace android
diff --git a/media/libstagefright/httplive/PlaylistFetcher.h b/media/libstagefright/httplive/PlaylistFetcher.h
index 2e28164f46..33b5879dc6 100644
--- a/media/libstagefright/httplive/PlaylistFetcher.h
+++ b/media/libstagefright/httplive/PlaylistFetcher.h
@@ -258,6 +258,8 @@ struct PlaylistFetcher : public AHandler {
     void updateDuration();
     void updateTargetDuration();
 
+    bool bufferAvailable() const;
+
     DISALLOW_EVIL_CONSTRUCTORS(PlaylistFetcher);
 };
 
